scripts/tuple_tree_generator/tuple_tree_generator/templates/struct.h.tpl

```diff
@@ -142,9 +157,54 @@ public:
   bool operator>(const /*= struct.name =*/ &Other) const { return key() > Other.key(); }
 
   /** else **/
-  bool operator==(const /*= struct.name =*/ &Other) const = default;
+  bool operator==(const /*= struct.name =*/ &Other) const {
+    /**- for field in struct.fields **/
+    if (/*= field.name =*/() != Other./*= field.name =*/())
+      return false;
+    /**- endfor **/
+    return true;
+  }
   /** endif **/
 
+  /**- if emit_tracking **/
+private:
+  template<size_t I>
+  revng::AccessTracker& getTracker() const {
+    /**- for field in struct.all_fields **/
+    if constexpr (I == /*= loop.index0 =*/)
+        return /*= field.name =*/Tracker;
+    /**- endfor -**/
+  }
+  /** if upcastable **/
+  /**- for child_type in upcastable|sort(attribute="user_fullname") **/
+  friend /*= child_type | fullname =*/;
+  /**- endfor **/
+  /** endif **//*# End UpcastablePointer stuff #*/
```

Comment:
Put "End UpcastablePointer stuff" on a new line

Reply:
i think i am going to just remove the comment, i copy pasted it from somewhere else and did not noticed it.